napari/layers/transforms.py

```diff
@@ -0,0 +1,173 @@
+
+    @property
+    def inverse(self):
+        if self._inverse_func is not None:
+            return Transform(self._inverse_func, self.func)
+        else:
+            raise ValueError('Inverse function was not provided.')
+
+
+# class TransformChain(ListModel, Transform):
+#     def __init__(self, transforms=[]):
+#         super().__init__(
+#             basetype=Transform,
+#             iterable=transforms,
+#             lookup={str: lambda q, e: q == e.name},
+#         )
+#
+#     def __call__(self, coords):
+#         return tz.pipe(coords, *self)
+#
+#     def __newlike__(self, iterable):
+#         return ListModel(self._basetype, iterable, self._lookup)
+#
+#     def set_slice(self, axes: Sequence[int]) -> TransformChain:
+#         return TransformChain([tf.set_slice(axes) for tf in self])
+
+
+class STTransform(Transform):
+    """n-dimensional scale and translation (shift) class.
+
+    Scaling is always applied before translation.
+
+    An empty translation vector implies no translation, and an empty scale
+    implies no scaling.
+
+    Translation is broadcast to 0 in leading dimensions, so that, for example,
+    a translation of [4, 18, 34] in 3D can be used as a translation of
+    [0, 4, 18, 34] in 4D without modification.
+
+    Translation is broadcast to 1 in leading dimensions, so that, for example,
+    a scale of [4, 18, 34] in 3D can be used as a scale of
+    [1, 4, 18, 34] in 4D without modification.
+    """
+
+    def __init__(self, scale=(1.0,), translate=(0.0,), name=None):
+        super().__init__(name=name)
+        self.scale = np.array(scale)
+        self.translate = np.array(translate)
+
+    def __call__(self, coords):
+        coords = np.atleast_2d(coords)
+        scale = np.concatenate(
+            ([1.0] * (coords.shape[1] - len(self.scale)), self.scale)
+        )
+        translate = np.concatenate(
+            ([0.0] * (coords.shape[1] - len(self.translate)), self.translate)
+        )
+        return np.squeeze(scale * coords + translate)
+
+    @property
+    def inverse(self):
+        return STTransform(1 / self.scale, -self.translate)
+
+    def set_slice(self, axes: Sequence[int]) -> STTransform:
+        return STTransform(self.scale[axes], self.translate[axes])
+
+    def set_pad(self, axes: Sequence[int]) -> STTransform:
+        n = len(axes) + len(self.scale)
+        not_axes = [i for i in range(n) if i not in axes]
+        scale = np.ones(n)
+        scale[not_axes] = self.scale
+        translate = np.zeros(n)
+        translate[not_axes] = self.translate
+        return STTransform(scale, translate)
+
+    def compose(self, transform: STTransform) -> STTransform:
+        scale = self.scale * transform.scale
+        translate = self.translate + self.scale * transform.translate
+        return STTransform(scale, translate)
+
+
+# class Translate(Transform):
+#     """n-dimensional translation (shift) class.
```

Comment:
Why are these classes commented out?

Reply:
We can delete if people don't think we need anymore, but I left them there as @jni put them in. I am in favor of deleting as I think the `STTransform` supersedes this one.

Reply:
Perhaps... See my comment above about STTransform and its inverse... =)

ST is a reasonably natural grouping but it seems to me to be more natural to leave them separate and manage them separately, as then you have a shot at reasoning about the order of operations...